xarray/tests/test_netcdftimeindex.py

```diff
@@ -0,0 +1,349 @@
+def test_get_slice_bound(date_type, index, kind):
+    result = index.get_slice_bound('0001', 'left', kind)
+    expected = 0
+    assert result == expected
+
+    result = index.get_slice_bound('0001', 'right', kind)
+    expected = 2
+    assert result == expected
+
+    result = index.get_slice_bound(
+        date_type(1, 3, 1), 'left', kind)
+    expected = 2
+    assert result == expected
+
+    result = index.get_slice_bound(
+        date_type(1, 3, 1), 'right', kind)
+    expected = 2
+    assert result == expected
+
+
+def test_date_type_property(date_type, index):
+    assert index.date_type == date_type
+
+
+def test_contains(date_type, index):
+    assert '0001-01-01' in index
+    assert '0001' in index
+    assert '0003' not in index
+    assert date_type(1, 1, 1) in index
+    assert date_type(3, 1, 1) not in index
+
+
+def test_groupby(da):
+    result = da.groupby('time.month').sum('time')
+    expected = xr.DataArray([4, 6], coords=[[1, 2]], dims=['month'])
+    assert_identical(result, expected)
+
+
+@pytest.mark.parametrize('sel_arg', [
+    '0001',
+    slice('0001-01-01', '0001-12-30'),
+    [True, True, False, False]
+], ids=['string', 'string-slice', 'bool-list'])
+def test_sel_string_or_list(da, index, sel_arg):
+    expected = xr.DataArray([1, 2], coords=[index[:2]], dims=['time'])
+    result = da.sel(time=sel_arg)
+    assert_identical(result, expected)
+
+
+def test_sel_date_slice_or_list(da, index, date_type):
+    expected = xr.DataArray([1, 2], coords=[index[:2]], dims=['time'])
+    result = da.sel(time=slice(date_type(1, 1, 1), date_type(1, 12, 30)))
+    assert_identical(result, expected)
+
+    result = da.sel(time=[date_type(1, 1, 1), date_type(1, 2, 1)])
+    assert_identical(result, expected)
+
+
+def test_sel_date_scalar(da, date_type, index):
+    expected = xr.DataArray(1).assign_coords(time=index[0])
+    result = da.sel(time=date_type(1, 1, 1))
+    assert_identical(result, expected)
+
+    result = da.sel(time='0001-01-01')
+    assert_identical(result, expected)
+
+
+def test_isel(da, index):
+    expected = xr.DataArray(1).assign_coords(time=index[0])
+    result = da.isel(time=0)
+    assert_identical(result, expected)
+
+    expected = xr.DataArray([1, 2], coords=[index[:2]], dims=['time'])
+    result = da.isel(time=[0, 1])
+    assert_identical(result, expected)
+
+
+@pytest.fixture
+def scalar_args(date_type):
+    return ['0001-01-01', date_type(1, 1, 1)]
+
+
+@pytest.fixture
+def range_args(date_type):
+    return ['0001', slice('0001-01-01', '0001-12-30'),
+            slice(None, '0001-12-30'),
+            slice(date_type(1, 1, 1), date_type(1, 12, 30)),
+            slice(None, date_type(1, 12, 30))]
+
+
+def test_indexing_in_series_getitem(series, index, scalar_args, range_args):
+    for arg in scalar_args:
+        assert series[arg] == 1
+
+    expected = pd.Series([1, 2], index=index[:2])
+    for arg in range_args:
+        pd.util.testing.assert_series_equal(series[arg], expected)
```

Comment:
Be careful. I don't think this is public API. Better to stick with the `.equals`.